### DISCUSSION TITLE
How to add Nuki devices as provided by Nuki Hub?

### DISCUSSION THREAD

**Discussion by Ra72xx** [andrew-codechimp/HA-Battery-Notes#2118] [General]
When using https://github.com/technyon/nuki_hub to access Nuki devices, there is exactly one MQTT device added to HA. This device has entities for all available Nuki devices, including their respective battery states, e.g. SmartLock (all kinds of version), Opener, Door Sensor, Keypad.
Is there a way to add Battery Notes information to these entities, too? It seems as if Battery Notes as of now assumes only one battery per device and does not handle a situation like this (one software device combining several physical devices)?

**Comment by andrew-codechimp** [comment]
Hi, there is a way to track these, I had similar issues with EcoWitt where all "sub-devices" were under 1 device in HA.
If you manually go and add a new battery note (Integrations/Add Integration/Battery Notes) and then choose Entity you can select the specific battery entity for the sub-device. You will probably want to give it a good name as you'll have several under 1 device. 
This will give you a battery type and quantity, battery last replaced and a battery-low for each sub-device, it won't give you a battery+ as that is tied at device level.
Mine were either showing voltage or a Low/Normal so I had to add a template as well which you can find out how to do [here](https://andrew-codechimp.github.io/HA-Battery-Notes/#battery-low-template) if you don't have percentages.

**Reply by RalphPungaKronbergs** [reply]
Hi, I do have the same setup for my EcoWitt sensors by manually adding all EcoWitt sensors as a single entity to Battery Notes. So far it seems to work fine using a template like `{{ states('sensor.gw1100a_soil_battery_4') | float(5) < 1.2 }}` but I can't figure how to use the entities manually entered name for the [Battery Low Notification](https://andrew-codechimp.github.io/HA-Battery-Notes/community/#battery-low-notification) script because `{{ trigger.event.data.device_name }}` is always replaced with the device name (e.g. "GW1100A") instead of the manually entered title for the entity. Any idea how to solve this?

Manually entered title for entity:

<img width="582" height="794" alt="image" src="https://github.com/user-attachments/assets/d1b7d92f-3f33-4d0e-a309-82d4e0e0d20f" />

.
But the resulting notification is using EcoWitt Hub's name instead:

<img width="488" height="251" alt="image" src="https://github.com/user-attachments/assets/f6bb5b2f-1273-428a-8347-6634a6ea8773" />

**Reply by andrew-codechimp** [reply]
Hi, that's not something I've handled and raises an interesting point.
At the moment I don't see an easy way to do this from a template, I will look into the best way to fix this in future. I'll raise an issue for myself so I don't forget this.

**Reply by RalphPungaKronbergs** [reply]
Thank you so much for having a look into this!

**Reply by andrew-codechimp** [reply]
An option you could use in the meantime is to change the template to be 
{{ trigger.event.data.device_name }} - {{ trigger.event.data.source_entity_id }}
It would at least give you a hint as to the appropriate sub-device that is having an issue.

**Comment by Ra72xx** [comment]
Thanks for the workaround! Seems a bit hacky, however. Probably it would be more elegant to optionally be able to select a battery entity directly from this integration?

**Reply by andrew-codechimp** [reply]
It's not really a work around, you can also go into the battery notes integration and choose add device which does the same, I just choose to explain it from the add integration method as it's easier to follow. 
It's exactly the same flow as adding a device that isn't in the library.
